sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobPrefix.java

```diff
@@ -16,14 +16,14 @@
      * The Name property.
      */
     @JsonProperty(value = "Name", required = true)
-    private String name;
+    private BlobName name;
 
     /**
      * Get the name property: The Name property.
      *
      * @return the name value.
      */
-    public String getName() {
+    public BlobName getName() {
```

Comment:
It's unfortunate this is a code generation change as it breaks the API. May need to look into have this generate as a new API and retain the older `String` variant. If you have questions or need help with this I've made code generation do this in Search in a few areas.

Reply:
Yea I was thinking that would probably be necessary. I can do what we did for BlobItemInternal and have BlobPrefixInternal in the implementation package and then turn BlobPrefix into a hand-written type